GeeksCoreLibrary/Components/Account/Services/AccountsService.cs

```diff
@@ -447,5 +454,199 @@ public async Task<List<RoleModel>> GetUserRolesAsync(ulong userId, bool includeP
+        /// </summary>
+        /// <param name="userIdForGoogleCid">The ID of the user to save the CID for.</param>
+        /// <param name="settings"></param>
+        public async Task SaveGoogleClientIdAsync(ulong userIdForGoogleCid, AccountCmsSettingsModel settings)
+        {
+            var request = httpContextAccessor.HttpContext?.Request;
+            var googleClientIdCookieValue = request?.Cookies[Constants.GoogleAnalyticsCookieName];
+            if (String.IsNullOrWhiteSpace(googleClientIdCookieValue))
+            {
+                return;
+            }
+
+            // GA1.2.1248174149.1587127355
+            var clientIdSplit = googleClientIdCookieValue.Split(new[] { '.' }, StringSplitOptions.RemoveEmptyEntries);
+
+            if (clientIdSplit.Length != 4)
+            {
+                var sanitizedGoogleClientIdCookieValue = googleClientIdCookieValue.Replace(Environment.NewLine, "").Replace("\n", "").Replace("\r", "");
+                logger.LogTrace($"Google Analytics cookie found ({Constants.GoogleAnalyticsCookieName}: {sanitizedGoogleClientIdCookieValue}), but is not a valid format. Expected to have 3 dots, but it has {clientIdSplit.Length - 1}", true);
+                return;
+            }
+
+            if (userIdForGoogleCid <= 0)
+            {
+                var sanitizedGoogleClientIdCookieValue = googleClientIdCookieValue.Replace(Environment.NewLine, "").Replace("\n", "").Replace("\r", "");
+                logger.LogTrace($"Google Analytics cookie found ({Constants.GoogleAnalyticsCookieName}: {sanitizedGoogleClientIdCookieValue}), but the user is not logged in and we did not create a new account (successfully).", true);
+                return;
+            }
+
+            var googleClientId = String.Join(".", clientIdSplit.Skip(2));
+            
+            
+            var wiserItemsService = ActivatorUtilities.CreateInstance<WiserItemsService>(serviceProvider, databaseConnection);
+            var tablePrefix = await wiserItemsService.GetTablePrefixForEntityAsync(settings.EntityType);
+
+            var detail = new WiserItemDetailModel()
+            {
+                Key = String.IsNullOrWhiteSpace(settings.GoogleClientIdFieldName) ? Constants.DefaultGoogleCidFieldName : settings.GoogleClientIdFieldName,
+                Value = googleClientId
+            };
+
+            await wiserItemsService.SaveItemDetailAsync(detail, userIdForGoogleCid, entityType: settings.EntityType);
+        }
+
+        /// <summary>
+        /// Saves a login attempt in the details of the user. This will use the query in <see cref="Settings.SaveLoginAttemptQuery"/>.
+        /// </summary>
+        /// <param name="success">Whether the login attempt was successful or not.</param>
+        /// <param name="userId">The ID of the user that is attempting to login.</param>
+        /// <param name="extraDataForReplacements"></param>
+        /// <param name="settings"></param>
```

Comment:
The summary can be replaced with `/// <inheritdoc />`, then the summary of the interface is shown so we only need to manage this on one place.